totalRP3_Extended_Tools/script/effects/effects.lua

```diff
@@ -740,6 +742,175 @@ local function speech_player_init()
+		else
+			table.insert(characterTableSorted[#characterTableSorted][2],  emote);
+		end
+	end
+
+	local currentCount = 0;
+	local currentTable;
+	local firstCharacter;
+	local previousCharacter;
+	for _, characterTable in ipairs(characterTableSorted) do
+		local character = characterTable[1];
+		local emotes = characterTable[2];
+		local characterSize = #emotes;
+		if (currentCount + characterSize) > maxSize then
+			-- Adding the previous table
+			if currentTable then
+				if firstCharacter ~= previousCharacter then
+					table.insert(output, { loc.EFFECT_DO_EMOTE_OTHER .. " " .. string.upper(firstCharacter) .. "-" .. string.upper(previousCharacter), currentTable });
+				else
+					table.insert(output, { loc.EFFECT_DO_EMOTE_OTHER .. " " .. string.upper(firstCharacter), currentTable });
+				end
+			end
+
+			if characterSize > maxSize then
+				-- Split into smaller tables
+				firstCharacter = nil;
+				local i = 0;
+				while (maxSize * i) < characterSize do
+					i = i + 1;
+					currentTable = {};
+					local offset = (i - 1) * maxSize;
+					for j = 1, min(maxSize, characterSize - offset) do
+						table.insert(currentTable, emotes[offset + j]);
+					end
+					table.insert(output, { loc.EFFECT_DO_EMOTE_OTHER .. " " .. string.upper(character) .. i, currentTable });
+				end
+				currentTable = nil;
+				currentCount = 0;
+			else
+				currentTable = {};
+				for i = 1, characterSize do
+					table.insert(currentTable, emotes[i]);
+				end
+				firstCharacter = character;
+				currentCount = characterSize;
+			end
+			previousCharacter = character;
+		else
+			if not currentTable then
+				currentTable = {};
+			end
+			for i = 1, characterSize do
+				table.insert(currentTable, emotes[i]);
+			end
+			currentCount = currentCount + characterSize;
+
+			if not firstCharacter then
+				firstCharacter = character;
+			end
+			previousCharacter = character;
+		end
+	end
+
+	-- Adding the previous table
+	if currentTable then
+		if firstCharacter ~= previousCharacter then
+			table.insert(output, { loc.EFFECT_DO_EMOTE_OTHER .. " " .. string.upper(firstCharacter) .. "-" .. string.upper(previousCharacter), currentTable });
+		else
+			table.insert(output, { loc.EFFECT_DO_EMOTE_OTHER .. " " .. string.upper(firstCharacter), currentTable });
+		end
+	end
+
+	return output;
+end
+
+local function do_emote_init()
+	local editor = TRP3_EffectEditorDoEmote;
+
+	-- Build list of emotes
+	local spokenEmotes = tInvert(TextEmoteSpeechList);
+	-- Those two are added dynamically
+	spokenEmotes["FORTHEALLIANCE"] = true;
+	spokenEmotes["FORTHEHORDE"] = true;
+	local animatedEmotes = tInvert(EmoteList);
+	local otherEmotes = {}
+	for i = 1, LAST_EMOTE_ID do
+		local emoteToken = _G["EMOTE" .. i .. "_TOKEN"]
+		if emoteToken then
+			if spokenEmotes[emoteToken] then
+				spokenEmotes[emoteToken] = i;
+			elseif animatedEmotes[emoteToken] then
+				animatedEmotes[emoteToken] = i
+			else
+				otherEmotes[emoteToken] = i;
+			end
+		end
+	end
```

Comment:
On second thought, you don't need a constant for `LAST_EMOTE_ID`.

```suggestion
	for _, emoteToken in pairs(hash_EmoteTokenList) do
		local tokenIndex = tonumber(string.match(emoteToken, "^EMOTE(%d+)_"));

		if spokenEmotes[emoteToken] then
			spokenEmotes[emoteToken] = tokenIndex;
		elseif animatedEmotes[emoteToken] then
			animatedEmotes[emoteToken] = tokenIndex;
		else
			otherEmotes[emoteToken] = tokenIndex;
		end
	end
```